server/controllers/request.js

```diff
@@ -0,0 +1,68 @@
+const mongoose = require('mongoose');
+const asyncHandler = require('express-async-handler');
+
+const Request = require('../models/Request');
+
+// @route GET /requests
+// @desc Get all requests of current users
+// @access Private
+exports.getRequests = asyncHandler(async (req,res,next) => {
+    try {
+        const requests = await Request.find({user: req.user.id}).populate('sitter');
+        res.status(200).json({requests});
+      } catch (error) {
+        res.status(500);
+        throw new Error(error.message);
+      }
+})
+
+// @route POST /requests
+// @desc Create new request for the user
+// @access Private
+exports.postRequest = asyncHandler(async (req,res,next) => {
+    const {sitter,start,end} = req.body;
+    const newRequest = new Request({
+        user: req.user.id,
+		sitter,
+		start,
+		end
+    });
+
+    try {
+        await newRequest.save();
+        res.status(201).json({request:newRequest});
+    } catch (error) {
+        res.status(500);
+        throw new Error(error.message);
+      }
+})
+
+// @route UPDATE /requests
+// @desc update request details
+// @access Private
+exports.updateRequest = asyncHandler(async (req,res,next) => {
+	const id = req.params.id;
+    const {sitter,start,end, accepted, declined, paid } = req.body;
```

Comment:
Including paid as an update option here is a security concern as a user could manually send a request to our API to update and say they've paid someone when they haven't. The only way to update paid should be within the actual pay route.

To a lesser extent, accepted and declined should be their own routes only accessible by the sitter, with this logic the owner could mark their own request as having been accepted before the sitter even sees it